internal/operator-controller/rukpak/convert/helm.go

```diff
@@ -17,12 +18,24 @@ type BundleToHelmChartConverter struct {
 	IsWebhookSupportEnabled bool
 }
 
-func (r *BundleToHelmChartConverter) ToHelmChart(bundle source.BundleSource, installNamespace string, watchNamespace string) (*chart.Chart, error) {
+func (r *BundleToHelmChartConverter) ToHelmChart(bundle source.BundleSource, config map[string]interface{}) (*chart.Chart, error) {
 	rv1, err := bundle.GetBundle()
 	if err != nil {
 		return nil, err
 	}
 
+	// Convert config map to render options
+	var opts []render.Option
+	if config != nil {
+		if installNs, ok := config[bundlepkg.BundleConfigInstallNamespaceKey].(string); ok {
+			opts = append(opts, render.WithInstallNamespace(installNs))
+		}
+		if watchNs, ok := config[bundlepkg.BundleConfigWatchNamespaceKey].(string); ok {
```

Comment:
let's bring back deriveDefaults and set watchNamespaces to:
- corev1.NamespaceAll iff AllNamespaces is supported
- install-namespace iff the only supported install-mode is OwnNamespace
- an error is its not set and SingleNamespace is the only supported mode
- an error if there are no supported install modes